Add kubeflow ART component (#206)
IBM/adversarial-robustness-toolbox: mlops/kubeflow/README.md

# Kubeflow Pipeline Components for ART

Kubeflow pipeline components are implementations of Kubeflow pipeline tasks. Each task takes
one or more [artifacts](https://www.kubeflow.org/docs/pipelines/overview/concepts/output-artifact/)
as input and may produce one or more
[artifacts](https://www.kubeflow.org/docs/pipelines/overview/concepts/output-artifact/) as output.

**Example: ART Components**

> * [Robustness Checker - PyTorch](robustness_checker_pytorch)

Reviewer: ```suggestion
* [Robustness Evaluation - FGSM - PyTorch](robustness_evaluation_fgsm_pytorch)
```

Each task usually includes two parts:

Each component has a component.yaml which will describe the functionality exposed by it, for e.g.

```

> name: 'PyTorch - Model Robustness Check'

Reviewer: ```suggestion
name: 'PyTorch - Fast Gradient Sign Method - Adversarial Robustness Evaluation'
```

description: |

>   Perform a robustness check using fast gradient attack with ART to make sure the model is robust against simple gradient changes.

Reviewer: ```suggestion
  Perform an evaluation of the adversarial robustness using the Fast Gradient Sign Method of the Adversarial Robustness Toolbox (ART).
```

  - {name: label_testset_path,           description: 'Required. Label test dataset path in the data bucket'}
  - {name: loss_fn,                      description: 'Required. PyTorch model loss function'}
  - {name: optimizer,                    description: 'Required. pytorch model optimizer'}
  - {name: clip_values,                  description: 'Required. pytorch model clip_values allowed for features (min, max)'}
  - {name: nb_classes,                   description: 'Required. The number of classes of the model'}
  - {name: input_shape,                  description: 'Required. The shape of one input instance for the pytorch model'}
  - {name: data_bucket_name,             description: 'Bucket that has the processed data',  default: 'training-data'}
  - {name: result_bucket_name,           description: 'Bucket that has the training results', default: 'training-result'}
outputs:
  - {name: metric_path,                  description: 'Path for robustness check output'}
  - {name: robust_status,                description: 'Path for robustness status output'}
implementation:
  container:

>     image: aipipeline/robustness-checker:pytorch

Reviewer: ```suggestion
    image: aipipeline/robustness-evaluation:pytorch
```

    command: ['python']
    args: [

>       -u, robustness_check.py,

Reviewer: ```suggestion
      -u, robustness_evaluation_fgsm_pytorch.py,
```

# Kubeflow Pipeline Components for ART

Kubeflow pipeline components are implementations of Kubeflow pipeline tasks. Each task takes
one or more [artifacts](https://www.kubeflow.org/docs/pipelines/overview/concepts/output-artifact/)
as input and may produce one or more
[artifacts](https://www.kubeflow.org/docs/pipelines/overview/concepts/output-artifact/) as output.

**Example: ART Components**
* [Adversarial Robustness Evaluation - FGSM - PyTorch](robustness_evaluation_fgsm_pytorch)

Each task usually includes two parts:

Each component has a component.yaml which will describe the functionality exposed by it, for e.g.

```
name: 'Adversarial Robustness Evaluation'
description: |

>   Perform an evaluation of the adversarial robustness using the Fast Gradient Sign Method of the Adversarial Robustness Toolbox (ART).

Reviewer 2: @Tomcli Please see my comment below on the disclaimer that I think we should add - it should go here, too.
